commit d21589bd97a5b6e76a2a291d063706700c8de5a6
Author:     Yan Cheng <58191769+yanchengnv@users.noreply.github.com>
AuthorDate: Thu Jun 26 17:06:21 2025 -0400
Commit:     GitHub <noreply@github.com>
CommitDate: Thu Jun 26 14:06:21 2025 -0700

    Consolidate admin port (#3519)
    
    Fixes # .
    
    ### Description
    
    This PR consolidates admin port number with the FL port number.
    
    - By default, the admin port is the same as the FL port.
    - User can still specify a separate admin_port in provision.
    - Admin Client is changed to use cellnet to communicate with the admin
    server.
    - Once logged in, the admin client does re-login automatically in case
    the admin server is restarted.
    - Enforce file integrity check on fed_admin.json. If modified, admin
    client won't start. User can change other params in
    local/resources.json.default in his startup kit.
    - Changed to use file streaming for submitting and downloading jobs.
    - Removed the use of overseer agent for admin client.
    - Changed FL Server to support multiple ports on the same cell.
    - Updated flare_api and FLAdminAPI.
    - Removed socket server testing, since we no longer use socket server
    for admin.
    - Make default scheme AIO based for server and relay.
    
    ### Types of changes
    <!--- Put an `x` in all the boxes that apply, and remove the not
    applicable items -->
    - [ ] Non-breaking change (fix or new feature that would not break
    existing functionality).
    - [x] Breaking change (fix or new feature that would cause existing
    functionality to change).
    - [ ] New tests added to cover the changes.
    - [ ] Quick tests passed locally by running `./runtest.sh`.
    - [ ] In-line docstrings updated.
    - [ ] Documentation updated.
    
    ---------
    
    Co-authored-by: Chester Chen <512707+chesterxgchen@users.noreply.github.com>
    Co-authored-by: nvkevlu <55759229+nvkevlu@users.noreply.github.com>
---
 nvflare/private/fed/client/client_engine.py | 9 ++++++++-
 nvflare/private/fed/server/server_engine.py | 8 +++++++-
 2 files changed, 15 insertions(+), 2 deletions(-)

diff --git a/nvflare/private/fed/client/client_engine.py b/nvflare/private/fed/client/client_engine.py
index dbf078c8a..b5067f064 100644
--- a/nvflare/private/fed/client/client_engine.py
+++ b/nvflare/private/fed/client/client_engine.py
@@ -266,6 +266,7 @@ class ClientEngine(ClientEngineInternalSpec, StreamableEngine):
         topic: str,
         factory: ConsumerFactory,
         stream_done_cb=None,
+        consumed_cb=None,
         **cb_kwargs,
     ):
         """Register a ConsumerFactory for specified app channel and topic.
@@ -281,6 +282,7 @@ class ClientEngine(ClientEngineInternalSpec, StreamableEngine):
             topic: app topic
             factory: the factory to be registered
             stream_done_cb: the callback to be called when streaming is done on receiving side
+            consumed_cb: the CB to be called after a chunk is consumed
 
         Returns: None
 
@@ -289,7 +291,12 @@ class ClientEngine(ClientEngineInternalSpec, StreamableEngine):
             raise RuntimeError("object streamer has not been created")
 
         self.object_streamer.register_stream_processing(
-            topic=topic, channel=channel, factory=factory, stream_done_cb=stream_done_cb, **cb_kwargs
+            topic=topic,
+            channel=channel,
+            factory=factory,
+            stream_done_cb=stream_done_cb,
+            consumed_cb=consumed_cb,
+            **cb_kwargs,
         )
 
     def shutdown_streamer(self):
diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index 49f1c20f3..b2d659daa 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -684,6 +684,7 @@ class ServerEngine(ServerEngineInternalSpec, StreamableEngine):
         topic: str,
         factory: ConsumerFactory,
         stream_done_cb=None,
+        consumed_cb=None,
         **cb_kwargs,
     ):
         if not self.run_manager:
@@ -693,7 +694,12 @@ class ServerEngine(ServerEngineInternalSpec, StreamableEngine):
             raise RuntimeError("object_streamer has not been created")
 
         self.run_manager.object_streamer.register_stream_processing(
-            channel=channel, topic=topic, factory=factory, stream_done_cb=stream_done_cb, **cb_kwargs
+            channel=channel,
+            topic=topic,
+            factory=factory,
+            stream_done_cb=stream_done_cb,
+            consumed_cb=consumed_cb,
+            **cb_kwargs,
         )
 
     def shutdown_streamer(self):
